garrysmod/lua/vgui/dcombobox.lua

```diff
@@ -123,7 +125,7 @@ function PANEL:OnSelect( index, value, data )
 
 end
 
-function PANEL:AddChoice( value, data, select, icon )
+function PANEL:AddChoice( value, data, select, icon, color, steamid64 )

```

Comment:
I don't like the 2 extra arguments. Can we have it be inferred from the type of the `icon` argument? i.e. if icon is a color - it's a color, if its an entity, grab the SteamID64, if it is a SteamID64, use that.

The only problematic one would be SteamID given that it's also a string, but I think it will be fine if we properly detect it's a SteamID.

Given this, this entire PR can be implemented in `PANEL.SetImage` alone.